profile/src/main/kotlin/com/imashnake/animite/profile/ProfileViewModel.kt

```diff
@@ -8,29 +9,40 @@ import com.imashnake.animite.core.data.Resource
 import com.imashnake.animite.core.data.Resource.Companion.asResource
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.flatMapLatest
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+@OptIn(ExperimentalCoroutinesApi::class)
 @HiltViewModel
 class ProfileViewModel @Inject constructor(
     private val userRepository: AnilistUserRepository,
-    private val preferencesRepository: PreferencesRepository
+    private val preferencesRepository: PreferencesRepository,
+    private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
+    private val refresh = savedStateHandle.getStateFlow(REFRESH, false)
```

Comment:
Should this really be stored in SavedStateHandle?

An alternative would be making a `MutableSharedFlow<Unit>`, which you can then `send` updates onto that would trigger data reloads

Reply:
We chose use this method in other parts of the app (search bar) so I'm going to stick with it for now.